node_modules/oae-core/documentpreview/js/documentpreview.js

```diff
@@ -71,6 +71,93 @@ define(['jquery', 'underscore', 'oae.core', 'lazyload'], function($, _, oae) {
         ////////////////////////////////
 
         /**
+         * Prepare markup for a page to be loaded
+         *
+         * @param  {Object}         page                Page object representing the page that will be loaded
+         */
+        var prepareForPageLoading = function(page) {
+            // Add the page container, containing a loading indicator
+            $spacer.before(oae.api.util.template().render($('#documentpreview-content-page-template', $rootel), {
+                'cssScopeClass': widgetData.previews.cssScopeClass,
+                'pageNumber': page.pageNumber
+            }));
+
+            // Cache a reference to the page element
+            page.$el = $('.documentpreview-content-page[data-page-number="' + page.pageNumber + '"]', $rootel);
+
+            // Cache the height and calculated top margin of the
+            // element as they are needed for zoom calculations
+            page.height = page.$el.height();
```

Comment:
In case the user scrolls the document while the page content is being retrieved. In that case the height is just the height of the loading message, but we still need to keep track of it.